assets/app/scripts/directives/buildTrendsChart.js

```diff
@@ -0,0 +1,252 @@
+"use strict";
+
+angular.module('openshiftConsole')
+  .directive('buildTrendsChart', function($filter, $location, $rootScope) {
+    return {
+      restrict: 'E',
+      scope: {
+        builds: '='
+      },
+      templateUrl: 'views/_build-trends-chart.html',
+      link: function($scope) {
+        var buildByNumber;
+        var completePhases = ['Complete', 'Failed', 'Cancelled', 'Error'];
+
+        // Simple humanize function that returns an abbreviated duration like "1h 4m" or "2m 4s"
+        // suitable for use on the chart y-axis.
+        var humanize = function(value) {
```

Comment:
instead of doing this here can you update the duration filter to optionally use abbreviated units?